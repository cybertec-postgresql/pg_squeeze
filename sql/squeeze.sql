CREATE EXTENSION pg_squeeze;

CREATE TABLE a(i int PRIMARY KEY, j int);

INSERT INTO a(i, j)
SELECT x, x
FROM generate_series(1, 10) AS g(x);

-- The trivial case.
SELECT squeeze.squeeze_table('public', 'a', NULL);

SELECT * FROM a;

-- Clustering by index.
CREATE INDEX a_i_idx_desc ON a(i DESC);
SELECT squeeze.squeeze_table('public', 'a', 'a_i_idx_desc');
SELECT * FROM a;

-- Involve TOAST.
CREATE TABLE b(i int PRIMARY KEY, t text);
INSERT INTO b(i, t)
SELECT x, repeat(x::text, 1024)
FROM generate_series(1, 10) AS g(x) GROUP BY x;
SELECT reltoastrelid > 0 FROM pg_class WHERE relname='b';
-- Copy the data into another table so we can check later.
CREATE TABLE b_copy (LIKE b INCLUDING ALL);
INSERT INTO b_copy(i, t) SELECT i, t FROM b;
-- Squeeze.
SELECT squeeze.squeeze_table('public', 'b', NULL);
-- Compare.
SELECT b.t = b_copy.t
FROM   b, b_copy
WHERE  b.i = b_copy.i;

