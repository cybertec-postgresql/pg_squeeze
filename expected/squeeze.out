CREATE EXTENSION pg_squeeze;
CREATE TABLE a(i int PRIMARY KEY, j int);
INSERT INTO a(i, j)
SELECT x, x
FROM generate_series(1, 10) AS g(x);
-- The trivial case.
SELECT squeeze.squeeze_table('public', 'a', NULL);
 squeeze_table 
---------------
 
(1 row)

SELECT * FROM a;
 i  | j  
----+----
  1 |  1
  2 |  2
  3 |  3
  4 |  4
  5 |  5
  6 |  6
  7 |  7
  8 |  8
  9 |  9
 10 | 10
(10 rows)

-- Clustering by index.
CREATE INDEX a_i_idx_desc ON a(i DESC);
SELECT squeeze.squeeze_table('public', 'a', 'a_i_idx_desc');
 squeeze_table 
---------------
 
(1 row)

SELECT * FROM a;
 i  | j  
----+----
 10 | 10
  9 |  9
  8 |  8
  7 |  7
  6 |  6
  5 |  5
  4 |  4
  3 |  3
  2 |  2
  1 |  1
(10 rows)

-- Involve TOAST.
CREATE TABLE b(i int PRIMARY KEY, t text);
INSERT INTO b(i, t)
SELECT x, repeat(x::text, 1024)
FROM generate_series(1, 10) AS g(x) GROUP BY x;
SELECT reltoastrelid > 0 FROM pg_class WHERE relname='b';
 ?column? 
----------
 t
(1 row)

-- Copy the data into another table so we can check later.
CREATE TABLE b_copy (LIKE b INCLUDING ALL);
INSERT INTO b_copy(i, t) SELECT i, t FROM b;
-- Squeeze.
SELECT squeeze.squeeze_table('public', 'b', NULL);
 squeeze_table 
---------------
 
(1 row)

-- Compare.
SELECT b.t = b_copy.t
FROM   b, b_copy
WHERE  b.i = b_copy.i;
 ?column? 
----------
 t
 t
 t
 t
 t
 t
 t
 t
 t
 t
(10 rows)

